packages/stream/get-stream/get-stream.js

```diff
@@ -0,0 +1,58 @@
+import { Internal } from '../../internal/index.js'
+
+/**
+ * @typedef FetchResponse
+ * @property {object} status -- A status response
+ * @property {object} data -- A return data as per endpoint
+ */
+
+/**
+ * A get specific stream data module based on the stream ID passing parameter
+ *
+ * @param {number} stream_id -- stream ID
+ * @returns {Promise<FetchResponse>} returns the restructured data which content status & specific stream data
+ * @throws {Error}
+ */
+export const getStream = async (stream_id) => {
+  if (stream_id === null || stream_id === undefined) {
+    throw new Error(
+      'Failed to get the stream data because the stream ID is empty. Please provide a stream ID'
+    )
+  } else if (typeof stream_id !== 'number') {
+    throw new TypeError(
+      'Failed to get the stream data because the stream ID is not in number format. A stream ID must be number format'
+    )
+  } else {
+    let fetchResponse = await Internal.fetchHttp({
+      url: `${Internal.config.api.base_url}/${Internal.config.api.version}/streams/${stream_id}`,
+      method: 'GET',
+    }).catch((error) => {
+      return error
```

Comment:
I think the only way to handle the error is by using a `catch` or `try catch` and not using a switch statement below. 

What I understand is every error always should be thrown using `throw` if you want to catch the error in another part of the code and should not be returned to a variable using `return` like here. But, if you don't have another catch in the code, the error should be handled here.

Reply:
@faiq-naufal for this one, I'm still thinking a way of it ya mas.. Because when I tried to use `try { await getStream()} catch {...}`, the nock testing for the positive case does not working. This error also occur if I add the `new Error` inside the `.catch`. Mock data on the nock testing said as `undefined`.

Reply:
@faiq-naufal pardon me, this one done at commit [cda642c](https://github.com/inlivedev/inlive-js-sdk/pull/12/commits/cda642c8c100a5796595a07ba12b4d75d7aa70ec). Kindly check mas. :)

Reply:
@faiq-naufal apologize mas, I would like to propose this one still using my initial codes (I revert back at commit [9dd4a03](https://github.com/inlivedev/inlive-js-sdk/pull/12/commits/9dd4a03183ce9cd7d5a63b337149a799167f33d5)).
This is because :
1. If I move the `throw new Error` inside `.catch`, the nock testing (positive case if reply 200), the mock data won't be readable. It will read as `undefined`.
2. Actually at first I add condition to check if data is `undefined` or not at testing, but it turned out making the positive case testing to be false green check pass.
3. I've tried so many ways including refactoring it using `try{ } catch { }` too (as per on internet examples), but it still turned out same (testing read the mock data as `undefined`).

After long exploring, it turns out that because using `throw new Error`.
Kindly read this reference : https://javascript.info/try-catch#throw-operator
On part in the article said :
> the throw operator generates a SyntaxError with the given message, the same way as JavaScript would generate it itself. The execution of try immediately stops and the control flow jumps into catch.
I think this is make sense why the Nock mock data read as `undefined` if I directly `throw new Error` inside `catch` (because due to the flow). 

Then I tried using` try { } catch { }` based on the article reference, and the testing worked fine again (not giving false green checks)!
But I prefer to use this initial codes because :
1. The flow almost same with the article (and the Nock not read as `undefined` so the testing just working fine).
2. The error message that the user can get based on the sentences what we write on `throw New Error`.

If any further advice is so welcome mas. I'm not sure if it's true like this, but so far from what I got (sudah mentok saya.. ^^")

Reply:
I see, I already tried your code and I also got the same `undefined` value in the test case when catching the error and throwing a new one. This is because the success response somehow is caught by the error `catch` although there shouldn't be an error. It feels weird, and I still don't know the cause. But, your implementation seems working fine so I think this is good for now.